lib/segment/src/segment_constructor/segment_constructor_base.rs

```diff
@@ -545,25 +597,27 @@ fn create_segment(
     let id_tracker = if mutable_id_tracker {
         // Determine whether we use the new (file based) or old (RocksDB) mutable ID tracker
         // Decide based on the feature flag and state on disk
-        let use_new_mutable_tracker = {
+        let use_new_mutable_tracker = if let Some(db) = db_builder.read() {
             // New ID tracker is enabled by default, but we still use the old tracker if we have
             // any mappings stored in RocksDB
+            //
             // TODO(1.15 or later): remove this check and use new mutable ID tracker unconditionally
```

Comment:
Not related to this PR, but it seems like we can't remove this check, as we never actually do migration to the new format. And even though, this function is called `create_segment`, it is also used during loading.